mlrun/serving/routers.py

```diff
@@ -278,7 +327,54 @@ def __init__(
             concurrent.futures.ProcessPoolExecutor,
             concurrent.futures.ThreadPoolExecutor,
         ] = None
-        super().__init__(**kwargs)
+
+    def _apply_logic(self, results: dict, event=None):
+        """
+        Apply merge logic on results.
+
+        :param results: A list of sample results by models e.g. results[model][prediction]
+        :param event: Response event
+
+        :return: Dictionary of results
+        """
+        if not self.extend_event:
+            event.body = {}
+        return self.merger(event.body, results)
+
+    def merger(self, body, results):
+        """Merging logic
+
+        input the event body and a dict of route results and returns a dict with merged results
+        """
+        for result in results.values():
+            body.update(result)
+        return body
```

Comment:
no need for both `merger` and `_apply_logic`. Simply copt the `merger` to where it is called (unless `merger` is used elsewhere in MLRun's code (not tests)).

Reply:
user face - we can change it but I don't think it necessary.